libusb/core.c

```diff
@@ -2277,6 +2279,26 @@ static enum libusb_log_level get_env_debug_level(void)
  * \see libusb_contexts
  */
 int API_EXPORTED libusb_init(libusb_context **ctx)
+{
+	return libusb_init_with_options(ctx, NULL, 0);
+}
+
+/** \ingroup libusb_lib
+ * Initialize libusb. This function must be called before calling any other
+ * libusb function.
+ *
+ * If you do not provide an output location for a context pointer, a default
+ * context will be created. If there was already a default context, it will
+ * be reused (and nothing will be initialized/reinitialized).
+ *
+ * \param ctx Optional output location for context pointer.
+ * Only valid on return code 0.
+ * \param options Optional array of options to set on the new context.
+ * \param num_options Number of elements in the options array.
+ * \returns 0 on success, or a LIBUSB_ERROR code on failure
+ * \see libusb_contexts
+ */
+int API_EXPORTED libusb_init_with_options(libusb_context **ctx, struct libusb_init_option options[], int num_options)
```

Comment:
Wouldn't making the array `const` be better?

Reply:
Done.